sql/src/main/java/io/crate/execution/dsl/phases/HashJoinPhase.java

```diff
@@ -90,6 +93,7 @@ public HashJoinPhase(StreamInput in) throws IOException {
 
         estimatedRowSizeForLeft = in.readVLong();
         numberOfRowsForLeft = in.readVLong();
+        rowsToBeConsumed = in.readInt();
```

Comment:
a streaming unit test would make sense, specially as the current impl is broken if `estimatedRowSizeForLeft` or `numberOfRowsForLeft` is negative